docs/Toy/toy/dialects/toy.py

```diff
@@ -87,6 +102,19 @@ def get_data(self) -> list[float]:
         return [float(el.value.data) for el in self.value.data.data]
 
 
+class InferAddOpShapeTrait(ToyShapeInferenceTrait):
+    @classmethod
+    def infer_shape(cls, op: Operation) -> None:
+        assert isinstance(op, AddOp)
```

Comment:
We don't have a way to enforce the type of the Op yet, but at this point it's programmer error to add the wrong trait to the op. It might be worth migrating to a ValueError, but that doesn't quite seem as appropriate as an assert.

Reply:
You can return a `TypeError` instead if you want, but I'm fine with an assert as well!